d2/src/main/java/com/linkedin/d2/balancer/strategies/degrader/D2MonitorEventEmitter.java

```diff
@@ -16,6 +16,7 @@
 
 package com.linkedin.d2.balancer.strategies.degrader;
 
+import com.linkedin.d2.balancer.clients.DegraderTrackerClient;
```

Comment:
This file should not be changed -- all needed function should be defined from the interface.

Reply:
You're right, more refactoring is needed. Let me do that.